Xamarin.Forms.Core.UnitTests/GridTests.cs

```diff
@@ -39,6 +39,164 @@ public void ThrowsOnNullRemove ()
 			Assert.Throws<ArgumentNullException> (() => layout.Children.Remove (null));
 		}
 
+		[TestFixture]
+		public class AddDimension : GridTests
+		{
+			[Datapoints]
+			public static IEnumerable<string> Operations = new[]
+			{
+				"HHH",
+				"HHV",
+				"HVH",
+				"HVV",
+				"VHH",
+				"VHV",
+				"VVH",
+				"VVV",
+
+				"RCRHVHVHVHVHV",
+
+				"HHHV",
+				"VVVH",
+
+				"RV",
+				"RH",
+				"CV",
+				"CH",
+
+				"RVRRV",
+				"CHCCH",
+			};
+
+			Grid _grid;
+
+			int _id = 0;
+			int _rowDef = 0;
+			int _colDef = 0;
+			int _totalWidth = 0;
+			int _totalHeight = 0;
+
+			public void AddHoizontal()
```

Comment:
If we're testing AddVertical then we we should test AddHorizontal. The tests are the same just transposed. More testing is a good thing. Helps prevent bugs like this one. :)